pkg/servercounter/lease_counter.go

```diff
@@ -0,0 +1,73 @@
+package servercounter
+
+import (
+	"fmt"
+	"time"
+
+	coordinationv1api "k8s.io/api/coordination/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	coordinationv1listers "k8s.io/client-go/listers/coordination/v1"
+	"k8s.io/klog/v2"
+)
+
+type ServerLeaseCounter struct {
+	lister        coordinationv1listers.LeaseLister
+	selector      labels.Selector
+	fallbackCount int
+}
+
+func NewServerLeaseCounter(lister coordinationv1listers.LeaseLister, labelSelector string, fallbackCount int) (*ServerLeaseCounter, error) {
+	selector, err := labels.Parse(labelSelector)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse label selector %v: %w", labelSelector, err)
+	}
+	return &ServerLeaseCounter{
+		lister:        lister,
+		selector:      selector,
+		fallbackCount: fallbackCount,
+	}, nil
+}
+
+func (lc *ServerLeaseCounter) CountServers() int {
+	leases, err := lc.lister.List(lc.selector)
```

Comment:
Could you add a comment here briefly describing the rational for not using a LIST and WATCH pattern (added complexity)?  It would help future readers of the code understand why we didn't do it initially

Reply:
Done.